extensions/indexes/range/src/org/exist/indexing/range/RangeIndexConfigAttributeCondition.java

```diff
@@ -0,0 +1,405 @@
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+import javax.xml.XMLConstants;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import static java.util.regex.Pattern.CASE_INSENSITIVE;
+
+/**
+ *
+ * A condition that can be defined for complex range config elements
+ * that compares an attribute.
+ *
+ * @author Marcel Schaeben
+ */
+public class RangeIndexConfigAttributeCondition extends RangeIndexConfigCondition{
+
+    private final String attributeName;
+    private final QName attribute;
+    private final String value;
+    private final Operator operator;
+    private boolean caseSensitive = true;
+    private boolean numericComparison = false;
+    private Double numericValue = null;
+    private String lowercaseValue = null;
+    private Pattern pattern = null;
+
+    public RangeIndexConfigAttributeCondition(Element elem, NodePath parentPath) throws DatabaseConfigurationException {
+
+        if (parentPath.getLastComponent().getNameType() == ElementValue.ATTRIBUTE) {
+            throw new DatabaseConfigurationException("Range index module: Attribute condition cannot be defined for an attribute:" + parentPath.toString());
+        }
+
+        this.attributeName = elem.getAttribute("attribute");
+        if (this.attributeName == null || this.attributeName.length() == 0) {
+            throw new DatabaseConfigurationException("Range index module: Empty or no attribute qname in condition");
+        }
+
+        this.attribute = new QName(QName.extractLocalName(this.attributeName), XMLConstants.NULL_NS_URI, QName.extractPrefix(this.attributeName), ElementValue.ATTRIBUTE);
+        this.value = elem.getAttribute("value");
+
+
+        // parse operator (default to 'eq' if missing)
+        if (elem.hasAttribute("operator")) {
+            final String operatorName = elem.getAttribute("operator");
+            this.operator = Operator.getByName(operatorName.toLowerCase());
+            if (this.operator == null) {
+                throw new DatabaseConfigurationException("Range index module: Invalid operator specified in range index condition: " + operatorName + ".");
+            }
+        } else {
+            this.operator = Operator.EQ;
+        }
+
+
+        final String caseString = elem.getAttribute("case");
+        final String numericString = elem.getAttribute("numeric");
+
+        this.caseSensitive = (caseString != null && !caseString.equalsIgnoreCase("no"));
+        this.numericComparison = (numericString != null && numericString.equalsIgnoreCase("yes"));
+
+        // try to create a pattern matcher for a 'matches' condition
+        if (this.operator == Operator.MATCH) {
+            final int flags = this.caseSensitive ? 0 : CASE_INSENSITIVE;
+            try {
+                this.pattern = Pattern.compile(this.value, flags);
+            } catch (PatternSyntaxException e) {
+                RangeIndex.LOG.error(e);
```

Comment:
Probably better to have `RangeIndex.LOG` as private and declare your own logger for this class.